complete/src/main/resources/static/app.js

```diff
@@ -230,20 +264,50 @@ function deleteRoom(room) {
 function showError(title, xhr) {
     var serverErrorMessage = xhr.responseJSON == null ? "No response from server." : xhr.responseJSON.message;
     console.error(title + "\n" + serverErrorMessage);
-    var notification = $("<div class=\"toast\" role=\"alert\" aria-live=\"assertive\" aria-atomic=\"true\" style=\"min-width: 30rem\">"
-            + "<div class=\"toast-header bg-danger\">"
-            + "<strong class=\"mr-auto text-dark\">Error</strong>"
-            + "<button type=\"button\" class=\"ml-2 mb-1 close\" data-dismiss=\"toast\" aria-label=\"Close\">"
-            + "<span aria-hidden=\"true\">&times;</span>"
-            + "</button>"
-            + "</div>"
-            + "<div class=\"toast-body\"><p>" + title + "</p><pre><code>" + serverErrorMessage + "</code></pre></div>"
-            + "</div>");
+    var notification = template({ title, serverErrorMessage }, props => `
+      <div class="toast" role="alert" aria-live="assertive" aria-atomic="true" style="min-width: 30rem">
+        <div class="toast-header bg-danger">
+          <strong class="mr-auto text-dark">Error</strong>
+          <button type="button" class="ml-2 mb-1 close" data-dismiss="toast" aria-label="Close">
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="toast-body">
+          <p>${props.title}</p>
+          <pre><code>${props.serverErrorMessage}</code><pre>
+        </div>
+    `);
     $("#notificationPanel").append(notification);
     notification.toast({delay: 30000});
     notification.toast('show');
 }
 
+function template(params, creator) {
+    return creator(sanatizeObject(params));
+}
+
+function sanatizeObject(object) {
```

Comment:
Looks like jquery has methods like html() and text() that do this for you already. No need to reinvent the wheel.

https://stackoverflow.com/questions/47219741/jquery-how-to-escape-html-to-prevent-xss/47219765

Reply:
I used them in the `toSafeHtml`; these two methods are to implement a very light-weight JSX syntax that auto-escapes strings for you.